src/sql/workbench/parts/dataExplorer/electron-browser/nodeActions.contribution.ts

```diff
@@ -15,27 +15,83 @@ MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 		id: DISCONNECT_COMMAND_ID,
 		title: localize('disconnect', 'Disconnect')
 	},
-	when: NodeContextKey.IsConnected
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnected'),
+		new ContextKeyEqualsExpr('isConnected', true))
 });
 
+// For Database nodes under a Server in the SQL Servers folder
 MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
 	group: 'connection',
 	order: 2,
 	command: {
 		id: NEW_QUERY_COMMAND_ID,
 		title: localize('newQuery', 'New Query')
 	},
-	when: NodeContextKey.IsConnectable
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnectable'),
+		new ContextKeyEqualsExpr('isConnectable', true),
+		new ContextKeyDefinedExpr('viewItem'),
+		new ContextKeyEqualsExpr('viewItem', 'Database'))
 });
 
+// For database nodes under the SQL Databases folder
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'connection',
+	order: 2,
+	command: {
+		id: NEW_QUERY_COMMAND_ID,
+		title: localize('newQuery', 'New Query')
+	},
+	when: ContextKeyExpr.and(
+		new ContextKeyDefinedExpr('isConnectable'),
+		new ContextKeyEqualsExpr('isConnectable', true),
+		new ContextKeyDefinedExpr('viewItem'),
+		new ContextKeyEqualsExpr('viewItem', 'azure.resource.itemType.database'))
```

Comment:
We shouldn't have azure specific strings in core. this means it won't work for cms